compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -406,6 +429,28 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
                     };
                     let ptr_imm = Immediate::new_pointer_with_meta(data, meta, &self.ecx);
                     ImmTy::from_immediate(ptr_imm, ty).into()
+                } else if matches!(kind, AggregateTy::Array) {
+                    if ty.layout.size().bytes() <= STACK_THRESHOLD {
+                        return None;
+                    }
+                    let mut mplace = None;
+                    let _alloc_id = self
+                        .ecx
+                        .intern_with_temp_alloc(ty, |ecx, dest| {
+                            // FIXME: Can we speed it up by using `ecx.write_immediate(.ScalarPair(_), dest)`?
+                            for (field_index, op) in fields.iter().copied().enumerate() {
+                                let field_dest = ecx.project_field(dest, field_index)?;
+                                ecx.copy_op(op, &field_dest)?;
+                            }
+
+                            let dest =
+                                dest.assert_mem_place().map_provenance(|prov| prov.as_immutable());
+                            mplace.replace(dest);
```

Comment:
Please don't. If you want to create a `mplace` here, you should build it from the returned `AllocId`, and avoid side-channelling the MemPlace.

Reply:
Could you point me to the related docs?
I don't figure out how to get a MemPlace or OpTy from `tcx.try_get_global_alloc()`.

Reply:
Okay, I figured it out. I should use `ecx.alloc` instead of `ecx.intern_with_temp_alloc` .
How fool I am :)